File: src/Notas.java
import javax.swing.*;
import java.util.Scanner;

public class Notas {
    Scanner ler = new Scanner(System.in);

    final double aprovado = 59.95;
    final double recuperacao = 59.94;
    final double reprovado = 29.94;
    int soma = 0;

    public void notas() {
        final double aprovado = 59.95;
        final double recuperacao = 59.94;
        final double reprovado = 29.94;
        String computado = " ";
        JOptionPane.showMessageDialog(null, "AVISO. So aceitamos notas maiores que 0 e menores  do que 20", "Erro", JOptionPane.ERROR_MESSAGE);

        for(int i = 1; i<=4 ; i++) {
            int nota;

            while (true) {
                nota = Integer.parseInt(JOptionPane.showInputDialog("Digite sua nota " + i + ":"));
                if (nota < 0 || nota > 20) {

                    JOptionPane.showMessageDialog(null, "Nota fora do intervalo de (a a 20). Tente novamente.");
                }
                else {
                    JOptionPane.showMessageDialog(null,"Nota computada");
                break;
                }

                soma += nota;

            }
        }
    }

    public void soma() {
        String somas = String.format("Sua nota eh %d ", soma);
        JOptionPane.showMessageDialog(null, somas);
    }

    public void resultado() {
        if (soma >= aprovado) {
            String aprovado = String.format("Voce foi aprovado!");
            JOptionPane.showMessageDialog(null, aprovado);

        } else if (soma <= recuperacao && soma > reprovado) {
            var notanecessaria = 120 - soma;
            String recuperacao = String.format("Voce esta de prova final!\nVoce precisa tirar na prova final %d pontos", notanecessaria);
            JOptionPane.showMessageDialog(null, recuperacao);
        } else if (soma <= reprovado) {
            String reprovado = String.format("Voce foi reprovado.");
            JOptionPane.showMessageDialog(null, reprovado);
        }

    }


}

File: Calculadora de Notas em Kotlin/src/main/kotlin/Notas.kt
import java.util.Scanner

fun Notas() {
    var ler = Scanner(System.`in`)
    val aprovado = 59.95
    val recuperacao = 59.94
    val reprovado = 29.94
    var soma = 0.0
    fun Digitarnotas() {
        println("\u001B[31mSo aceitamos notas maiores que 0 e menores que 20 \u001B[0m")
        println("\u001B[31mPOR FAVOR DIGITE O VALOR COM VIRGULA NAO COM PONT0 \u001B[0m")
        for (i in 1..4) {
            var notas = 0.0
            while (true) {
                var nota = print("Digite sua $i -> ")
                notas = ler.nextDouble()
                val errodigitacao = notas?.toFloat() ?: 0.0
                if (notas < 0.0 || notas > 20.0) {
                    println("Nota fora do intervalor. Digite novamente")
                } else {
                    println("Nota Computada")
                    soma += notas
                    break
                }

            }
        }
    }

    Digitarnotas()

    fun imprimirsoma() {
        println("Sua nota eh $soma")
    }
    imprimirsoma()
    fun resultado() {
        if (soma >= aprovado) {
            println("\u001B[32mVoce foi aprovado. \u001B[0m ")
        } else if (soma <= recuperacao && soma > reprovado) {
            var notaparafinal = 120 - soma
            println("\u001B[33mVoce esta de prova final. \u001B[0m")
            println("Tire na prova nota igual ou superior a $notaparafinal para ser aprovado")
        } else if (soma <= reprovado) {
            println("\u001B[31mVoce esta reprovado \u001B[0m")
        } else {
            println("Nao sei o que dizer")
        }


    }
    resultado()

}